fe1-web/src/core/network/NetworkManager.ts

```diff
@@ -139,7 +139,10 @@ class NetworkManager {
 
     if (alwaysPersistConnection) {
       // always push the connection to the array so that we can re-connect later
-      this.connections.push(connection);
+      // if it doesn't already exist to avoid duplicates
```

Comment:
Great improvement! Do you think it makes sense to use a set instead of an array?

Reply:
Arrays in Typescript have useful functions that Sets do not. 
I see that sets can be useful because of the uniqueness of their elements, but due to the above point, I'd say we keep it as an array. I don't think that checking if a connection already exists in the array is going to introduce much overhead considering that the array is probably not going to be that big.
What do you think?